keps/sig-apps/4017-pod-index-label/README.md

```diff
@@ -0,0 +1,898 @@
+Indexed completion mode). The details of this are outlined in the Design Details section below.
+
+StatefulSet pod label: `statefulset.kubernetes.io/pod-ordinal`
+Indexed Job pod label: `batch.kubernetes.io/job-completion-index` (same as existing annotation)
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+
+#### Story 1
+As a user, I would like to lookup a job's pod logs by its index.
+
+#### Story 2
+As a user, I would like to target traffic to a specific pod index (e.g., index 0) in a StatefulSet.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+The StatefulSet controller will only need a minor update to the [newStatefulSetPod](https://github.com/kubernetes/kubernetes/blob/fb5e9ef3b2a6f2136d54868187431c345e59f55f/pkg/controller/statefulset/stateful_set_utils.go#L458) function, to set the pod ordinal as the label `statefulset.kubernetes.io/pod-ordinal`. This call is downstream from the [newVersionedStatefulSetPod](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/stateful_set_control.go#L416) call, which generates
+the StatefulSet pods before creating them as necessary in [CreateStatefulPod](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/stateful_set_control.go#L433).
+
+Similarly, the Job controller would need to add the completion index as a label [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/job_controller.go#L1481)
+where it adds the corresponding annotation.
+
+Next, we must determine how enabling this new feature will interact with existing workloads. There are a couple of options:
+
+1. Only inject the label on *newly created pods*, so an existing StatefulSet/Indexed Jobs may include pods with the label and some without it. 
```

Comment:
I was told that when you have unresolved items it sometimes recommended to wrap them like this?

```
<<[UNRESOLVED optional short context or usernames ]>>
Stuff that is being argued.
<<[/UNRESOLVED]>>
```
It sounds like you are asking the reviewers to decide on design 1 or design 2 so you may want to make it clear that this is still a question?

Reply:
Added! thanks for the tip